uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Task.php

```diff
@@ -63,4 +63,115 @@ private static function getPreviousAssigneeId($taskId) {
 
     return $id;
   }
+
+  /**
+   * Does user has sufficient permission for view/edit activity record.
+   *
+   * @param int $activityId
+   *   Activity record id.
+   * @param int $action
+   *   Edit/view.
+   *
+   * @return bool
+   */
+  public static function checkPermission($activityId, $action) {
+    $allow = FALSE;
+    if (!$activityId ||
+      !in_array($action, array(CRM_Core_Action::UPDATE, CRM_Core_Action::VIEW))
+    ) {
+      return $allow;
+    }
+
+    $activity = new CRM_Activity_DAO_Activity();
+    $activity->id = $activityId;
+    if (!$activity->find(TRUE)) {
+      return $allow;
+    }
+
+    // Component related permissions.
+    $compPermissions = array(
+      'CiviTask' => array('access Tasks and Assignments'),
+      'CiviDocument' => array('access Tasks and Assignments'),
+    );
+
+    // First check the component permission.
+    $sql = "
```

Comment:
is it possible to use the API instead of sql here .. also maybe the logic to get component permissions should be separated into separate method.